portfolio/src/main/webapp/script.js

```diff
@@ -29,10 +36,14 @@ hideables.forEach(hideable => {
   h3.addEventListener('click', () => ul.classList.toggle("hidden"));
 });
 
-function validateComment() {
-  const name = document.getElementById("name").value;
+function validateComment(name, text) {
   const regex = "^[a-zA-Z0-9]+([_.-]?[a-zA-Z0-9])*$";
 
+  if (user == null) {
+    alert("You must sign in to submit a comment.");
+    return false;
+  }
+  
   if (!name.match(regex)) {
     alert("Invalid name");
```

Comment:
(not related to this pull request, but I have this thought now) give some hints regarding what should a good name look like

Reply:
(feel free to ignore this comment)